pkg/remote/crd_upgrade.go

```diff
@@ -21,6 +29,142 @@ func PatchCRD(ctx context.Context, clnt client.Client, crd *v1extensions.CustomR
 		client.FieldOwner(v1beta2.OperatorName))
 }
 
+type CrdType string
+
+const (
+	KCP CrdType = "KCP"
+	SKR CrdType = "SKR"
+)
+
+func updateRemoteCRD(
+	ctx context.Context, plural string, kyma *v1beta2.Kyma, runtimeClient Client, controlPlaneClient Client) (
+	bool, *v1extensions.CustomResourceDefinition, *v1extensions.CustomResourceDefinition, error) {
+	crd := &v1extensions.CustomResourceDefinition{}
+	crdFromRuntime := &v1extensions.CustomResourceDefinition{}
+	var err error
+	err = controlPlaneClient.Get(
+		ctx, client.ObjectKey{
+			// this object name is derived from the plural and is the default kustomize value for crd namings, if the CRD
+			// name changes, this also has to be adjusted here. We can think of making this configurable later
+			Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group),
+		}, crd,
+	)
+
+	if err != nil {
+		return false, nil, nil, err
+	}
+
+	err = runtimeClient.Get(
+		ctx, client.ObjectKey{
+			Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group),
+		}, crdFromRuntime,
+	)
+
+	if ShouldPatchRemoteCRD(crdFromRuntime, crd, kyma, err) {
+		err = PatchCRD(ctx, runtimeClient, crd)
+		if err != nil {
+			return false, nil, nil, err
+		}
+
+		err = runtimeClient.Get(
+			ctx, client.ObjectKey{
+				Name: fmt.Sprintf("%s.%s", plural, v1beta2.GroupVersion.Group),
+			}, crdFromRuntime,
+		)
+		if err != nil {
+			return false, nil, nil, err
+		}
+
+		return true, crdFromRuntime, crd, nil
+	}
+
+	if err != nil {
+		return false, nil, nil, err
+	}
+
+	return false, nil, nil, nil
+}
+
+func ShouldPatchRemoteCRD(
+	runtimeCrd *v1extensions.CustomResourceDefinition, kcpCrd *v1extensions.CustomResourceDefinition,
+	kyma *v1beta2.Kyma, err error) bool {
+	if k8serrors.IsNotFound(err) {
```

Comment:
don't put err as input

Reply:
We no longer need this check since we do the generation check separately